source/data-formats/dates-and-times.txt

```diff
@@ -0,0 +1,448 @@
+  and stores it to MongoDB unchanged.
+- If the ``datetime`` is aware, {+driver-short+} automatically
+  converts the time to UTC before storing it to MongoDB.
+
+The following code example inserts a document containing a ``datetime`` value localized
+to the ``"US/Pacific"`` time zone. {+driver-short+} uses the attached time zone to convert
+the local time to UTC. When the document is retrieved from MongoDB, the ``datetime`` value
+is in UTC.
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: python
+
+      from pymongo import MongoClient 
+      from datetime import datetime
+      from bson.codec_options import CodecOptions
+      import pytz
+      from pytz import timezone
+      
+      utc_datetime = datetime(2002, 10, 27, 6, 0, 0)
+      pacific = timezone("US/Pacific")
+      local_datetime = pacific.localize(utc_datetime)
+
+      print(f"datetime before storage: {local_datetime}")
+      collection.insert_one({"date": local_datetime}) 
+      find_result = collection.find_one()["date"]
+      print(f"datetime after storage: {find_result}")
+
+   .. output::
+      :language: shell
+      :visible: false
+
+      datetime before storage: 2002-10-27 06:00:00-08:00
+      datetime after storage: 2002-10-27 14:00:00
+
+.. important:: datetime.now()
+   
+   Avoid calling the ``datetime.now()`` method, with no arguments, which returns the
+   current local time.
+   
+   Instead, always call the ``datetime.now(tz=datetime.timezone.utc)`` method, which
+   returns the current time in UTC.
+
+.. _pymongo-out-of-range-datetimes:
+
+Handling Out-of-Range datetimes
+-------------------------------
+
+Python's ``datetime`` class can represent only ``datetime`` values between
+``datetime.min`` and ``datetime.max`` (years 1-9999).
+You can represent a much greater range of dates and times by using BSON, which allows any
+64-bit millisecond value from the Unix epoch. 
+
+To represent a BSON time with {+driver-short+}, create a
+``datetime_ms.DatetimeMS`` object, a wrapper for Python's built-in ``int`` type.
+You can manually create a ``DatetimeMS`` object by passing in one of the following values:
+
+- An ``int`` representing the number of milliseconds since the Unix epoch
+- A ``datetime`` object
+
+The following code example constructs a ``DatetimeMS`` object by passing in an ``int``
+value that represents the year -146136543, a date outside of the ``datetime`` range: 
+
+.. code-block:: python
+
+     from bson.datetime_ms import DatetimeMS
+
+     out_of_range = DatetimeMS(-(2**62))
+
+You can also instruct {+driver-short+} to automatically decode UTC ``datetime`` values as
+``DatetimeMS`` objects. To do so, set the ``datetime_conversion`` parameter of ``CodecOptions``
+to a value from the ``datetime_ms.DatetimeConversion`` enum. The following sections
+describe these values.
+
+.. tip::
+
+   For more information about the Unix epoch, see its :wikipedia:`Wikipedia <Unix_time>`
+   entry.
```

Comment:
S: I think this should be moved closer to the first mention of "Unix epoch." I spent this whole section wondering what that meant so I didn't have as much context as I probably should have

Reply:
Sorry! I initially had it inline in the section intro, but thought it was more consistent with the SG and other docs to put it in an admonition like this. Happy to go back to the way it was.